spec/catalog.yaml

```diff
@@ -59,4 +59,205 @@ paths:
+
+  /v1/databases:
+    post:
+      tags: [ Database ]
+      summary: >
+        Create a new database. A catalog can manage one or more databases. A database is used to manage one or more tables.
+        There are three modes when trying to create a database:
+          * CREATE: create with fast fail if there is already a database existing with the same name.
+          * EXIST_OK: create and ignore existing database with the same name.
+          * OVERWRITE: it's a dangerous mode, it will delete the existing database (and all the assets it manages), then create a new one.
+      operationId: CreateDatabase
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDatabaseRequest'
+      responses:
+        200:
+          $ref: '#/components/responses/CreateDatabaseResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        406:
+          $ref: '#/components/responses/UnsupportedOperationResponse'
+        503:
+          $ref: '#/components/responses/ServiceUnavailableResponse'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+components:
+
+  schemas:
+    ErrorModel:
+      type: object
+      description: JSON error payload returned in a response with further details on the error
+      required:
+        - message
+        - type
+        - code
+      properties:
+        message:
+          type: string
+          description: Human-readable error message
+        type:
+          type: string
+          description: Internal type definition of the error
+          example: NoSuchDatabaseException
+        code:
+          type: integer
+          minimum: 400
+          maximum: 600
+          description: HTTP response code
+          example: 404
+        stack:
+          type: array
+          items:
+            type: string
+
+    LanceCatalogErrorResponse:
+      description: JSON wrapper for all error responses (non-2xx)
+      type: object
+      required:
+        - error
+      properties:
+        error:
+          $ref: '#/components/schemas/ErrorModel'
+      additionalProperties: false
+      example:
+        {
+          "error": {
+            "message": "The server does not support this operation",
+            "type": "UnsupportedOperationException",
+            "code": 406
+          }
+        }
+
+    CreateDatabaseRequest:
+      type: object
+      required: [ name, mode ]
+      properties:
+        name: { type: string }
+        mode:
+          type: string
+          enum: [ CREATE, EXIST_OK, OVERWRITE ]
+        options:
+          type: object
+          additionalProperties: { type: string }
+
+    CreateDatabaseResponse:
+      type: object
+      required:
+        - database
+      properties:
+        database: { type: string }
```

Comment:
I think this should be `name`? to be consistent with the `CreateDatabaseRequest`